trieste/bayesian_optimizer.py

```diff
@@ -449,6 +492,8 @@ def optimize(
             :class:`Record`.
         :param track_state: If `True`, this method saves the optimization state at the start of each
             step. Models and acquisition state are copied using `copy.deepcopy`.
+        :param track_path: If set, the optimization state is saved to disk at this path,
```

Comment:
I would add a note in the class docstring as well, expanding a bit, e.g. indicating when this would be desirable